modules/mobile/spec/request/letters_request_spec.rb

```diff
@@ -169,159 +122,68 @@
-            allow(InvalidLetterAddressEdipi).to receive(:find_or_create_by).and_raise(ActiveRecord::ActiveRecordError)
-            expect { get '/mobile/v0/letters', headers: iam_headers }.to change(InvalidLetterAddressEdipi, :count).by(0)
-            expect(response).to have_http_status(:unprocessable_entity)
-          end
+          expect(response).to have_http_status(:internal_server_error)
+          expect(response.parsed_body).to eq({ 'errors' =>
+                                                 [{ 'code' => '500',
+                                                    'source' => 'Lighthouse::LettersGenerator::Service',
+                                                    'status' => '500',
+                                                    'meta' => { 'message' => nil } }] })
         end
       end
     end
 
-    context 'with a not eligible error' do
-      it 'returns a not found response' do
-        VCR.use_cassette('evss/letters/letters_not_eligible_error') do
-          get '/mobile/v0/letters', headers: iam_headers
-          expect(response).to have_http_status(:bad_gateway)
-          expect(response.body).to match_json_schema('evss_errors')
-          expect(JSON.parse(response.body)).to have_deep_attributes(
-            'errors' => [
-              {
-                'title' => 'Proxy error',
-                'detail' => 'Upstream server returned not eligible response',
-                'code' => '111',
-                'source' => 'EVSS::Letters::Service',
-                'status' => '502',
-                'meta' => {
-                  'messages' => [
-                    {
-                      'key' => 'lettergenerator.notEligible',
-                      'severity' => 'FATAL',
-                      'text' => 'Veteran is not eligible to receive the letter'
-                    }
-                  ]
-                }
-              }
-            ]
-          )
+    context 'with upstream service error' do
+      it 'returns a internal server error response' do
+        VCR.use_cassette('mobile/lighthouse_letters/letters_500_error_bgs', match_requests_on: %i[method uri]) do
+          get '/mobile/v0/letters/beneficiary', headers: iam_headers
+          expect(response).to have_http_status(:internal_server_error)
+          error = response.parsed_body['errors']
+          expect(error).to eq([{ 'title' => 'Required Backend Connection Error',
+                                 'detail' => 'Required Backend Connection Error',
+                                 'code' => '500',
+                                 'source' => 'Lighthouse::LettersGenerator::Service',
+                                 'status' => '500',
+                                 'meta' => { 'message' => 'Backend Service Error BGS' } }])
         end
       end
     end
 
-    context 'with can not determine eligibility error' do
+    context 'with one or more letter destination errors' do
```

Comment:
What makes this a letter destination error? The vcr cassette seems to be 404 not found.

Reply:
This test already existed in the old LH specs file. I think it is for when a record identified by an ICN number can't be found

Reply:
It's true that a test with this description existed in the old specs, but I don't think it's testing the case where an ICN can't be found. If you look at that cassette, you'll see that it returns 500 with messages including:
```
{
  "key": "letterDestination.city.invalid",
  "severity": "FATAL",
  "text": "City is invalid"
}
```


That same response may exist from lighthouse but we don't have a cassette for it, and I don't think we really need to because of the way the lighthouse service is handling the errors. If you look at the `Lighthouse::LettersGenerator::ServiceError::ERROR_MAP`, you'll see that any error status other than the ones listed will default to 500, which is why the service unavailable test on line 160 above returns 500 despite the upstream returning 503. I think the `returns a internal server error response` test on line 175 above actually covers the `letter destination error` functionality because it shows that any 500 from upstream will return a 500 downstream. 

The cassette you're using here is for the case where the ICN is not found, and this is a valid test. But the description is not accurate.

Reply:
got it